### PR TITLE
Signals: `submitSignal` typing -- update `content` to `unknown`

### PR DESCRIPTION
**Pull request by WillieHabi**
## Description
[ADO Task 7025 ](https://dev.azure.com/fluidframework/internal/_workitems/edit/7025)

Follow up work to https://github.com/microsoft/FluidFramework/pull/17729

Updating `submitSignal` `content` arg typing. It should be `unknown` instead of `any` / `IDocumentMessage`.

## Breaking Changes
This PR includes several type changes to public and alpha API. 

## Reviewer Guidance
- Refer to original PR for context regarding this follow up change

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts [hunk 1]
```diff
@@ -723,7 +723,7 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
 		return !this.disposed && this.socket.connected;
 	}
 
-	protected emitMessages(type: string, messages: IDocumentMessage[][]) {
+	protected emitMessages(type: string, messages: unknown[] | unknown[][]) {
 		// Only submit the op/signals if we are connected.
```
Comment (jason-ha) [inline packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts:727]: This doesn't feel right.
`IDocumentMessage` is much nicer than unknown. Is there something else that should be used? Something for signals and leave IDocMsg for ops?

File: packages/drivers/driver-base/src/documentDeltaConnection.ts [hunk 2]
```diff
@@ -308,7 +308,7 @@ export class DocumentDeltaConnection
 		return this.details.initialClients;
 	}
 
-	protected emitMessages(type: string, messages: IDocumentMessage[][]) {
+	protected emitMessages(type: string, messages: unknown[][]) {
 		// Although the implementation here disconnects the socket and does not reuse it, other subclasses
```
Comment (jason-ha) [inline packages/drivers/driver-base/src/documentDeltaConnection.ts:312]: This is not good for ops case. Is there something better?

Comment (WillieHabi) [reply packages/drivers/driver-base/src/documentDeltaConnection.ts:312]: (IDocumentMessage | unknown)[][] ?

Comment (WillieHabi) [reply packages/drivers/driver-base/src/documentDeltaConnection.ts:312]: I actually see Gary used `(IDocumentMessage | unknown)[][] | ISentSignalMessage[]` in his original commit -- I'll change to this for now

Comment (jason-ha) [reply packages/drivers/driver-base/src/documentDeltaConnection.ts:316]: `unknown` still seems like a problem. Isn't `Foo | unknown` just `unknown`?

Comment (WillieHabi) [reply packages/drivers/driver-base/src/documentDeltaConnection.ts:316]: Yeah you're right it just becomes `unknown`. I think the solution is `IDocumentMessage[][] | ISentSignalMessage[]` and leave `content: IDocumentMessage` in submitSignal signature

Comment (WillieHabi) [reply packages/drivers/driver-base/src/documentDeltaConnection.ts:313]: Yes that's right -- will insted use `(IDocumentMessage[] | ISentSignalMessage)[]`

**Pull request closed without merging**